cmd/goal/node.go

```diff
@@ -164,9 +167,18 @@ var catchupCmd = &cobra.Command{
 				URL := fmt.Sprintf(catchpointURL, genesis)
 				label, err := getMissingCatchpointLabel(URL)
 				if err != nil {
-					reportErrorf(errorCatchpointLabelMissing, errorUnableToLookupCatchpointLabel)
+					reportErrorf(errorCatchpointLabelMissing, errorUnableToLookupCatchpointLabel, err.Error())
 				}
 				args = append(args, label)
+				if !fastCatchupForce {
+					fmt.Printf(nodeConfirmImplicitCatchpoint, label)
+					reader := bufio.NewReader(os.Stdin)
+					text, _ := reader.ReadString('\n')
+					text = strings.Replace(text, "\n", "", -1)
+					if text != "yes" {
+						reportErrorf(errorAbortedPerUserRequest)
```

Comment:
would this be worth being a utility function for "get user input"?

Reply:
In the cmd directory we call `ReadString` in 5 places, there's probably some duplicate code there. Sounds like a fun little refactor, you should make a PR.